### PR TITLE
Remove old regress tests

### PR DESCRIPTION
**Pull request by joergsteffens**
The tests from `regress/` have been deprecated in favor the `systemtests/`.
As they  are no longer is use, this PR removes them.
A couple of useful tests are migrated to systemtests.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [ ] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [ ] `check-sources --since-merge` does not report any problems
- [ ] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

File: systemtests/scripts/functions [hunk 1]
```diff
@@ -528,17 +481,41 @@ change_files()
...
    SIZE=-1
    if [ -e "$FILE" ]; then
-     SIZE="$(du "$FILE" | cut -f 1)"
+     SIZE="$(du -B 1 --apparent-size "$FILE" | cut -f 1)"
    fi
    print_debug "$FILE: $SIZE bytes"
    echo "$SIZE"
 }
 
+# get_file_size()
```
Comment (pstorz) [inline systemtests/scripts/functions:507]: If this block is not needed anymore, why not remove completely?

File: systemtests/scripts/create_sparse_file.sh [hunk 2]
```diff
@@ -26,12 +31,14 @@ echo "start" > $filename
 dd if=/dev/zero of=$filename bs=1 count=0 seek=$size 2>/dev/null
 echo "end" >> $filename
 
-size="`du --block-size=1 --apparent-size ${filename} | cut -f 1`"
-realsize="`du --block-size=1 ${filename} | cut -f 1`"
+size="$(du -B 1 --apparent-size ${filename} | cut -f 1)"
```
Comment (pstorz) [inline systemtests/scripts/create_sparse_file.sh:34]: FreeBSD *du* does not support long options

File: systemtests/tests/python-bareos/testrunner [hunk 3]
```diff
@@ -18,8 +18,8 @@ JobName=backup-bareos-fd
 
 
 # Directory to backup.
-# This directory will be created by setup_data "$@"().
-BackupDirectory="${tmp}/data"
+# This directory will be created by setup_data().
+export BackupDirectory="${tmp}/data"
```
Comment (pstorz) [inline systemtests/tests/python-bareos/testrunner:22]: Is this not already done now in environment Line 21?

File: systemtests/tests/block-size/testrunner [hunk 4]
```diff
@@ -1,53 +1,47 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (pstorz) [inline systemtests/tests/block-size/testrunner:1]: What is this block-size test needed for? The block size changes usually only make sense on tape drives. and this test is not run against a tape drive here.

File: systemtests/tests/fileset-multiple-include-blocks/testrunner [hunk 5]
```diff
@@ -1,15 +1,13 @@
-#!/bin/sh
-
-. scripts/functions
+#!/bin/bash
```
Comment (pstorz) [inline systemtests/tests/fileset-multiple-include-blocks/testrunner:1]: Probably this test and the multiple-options-blocks tests can be merged into one test.

**Review by pstorz — CHANGES_REQUESTED** [review]
Good work. It might make sense to see if we can merge some of the new tests or /and also of the existing tests together to get a better overall run performance. By using setup and cleanup steps

File: systemtests/tests/python-bareos/testrunner (hunk 3 above)
Comment (joergsteffens) [reply systemtests/tests/python-bareos/testrunner]: Rmeoved line.

File: systemtests/tests/block-size/testrunner (hunk 4 above)
Comment (joergsteffens) [reply systemtests/tests/block-size/testrunner:1]: Modified to be a tape test.

File: systemtests/tests/fileset-multiple-include-blocks/testrunner (hunk 5 above)
Comment (joergsteffens) [reply systemtests/tests/fileset-multiple-include-blocks/testrunner:1]: Yes. As discussed, we first integrate this like this and in a separate PR we try to reduce the number of full tests by cmake FIXTURES.

**Comment by joergsteffens** [comment]
This PR depends on #809 and should be merged after that.

**Comment by joergsteffens** [comment]
I think, I covered all requested changes. Also PR #809 is merged now, so this PR is rebased on it.
@pstorz : please review again.

**Review by pstorz — CHANGES_REQUESTED** [review]
- CHANGELOG.md is missing, I think that would make sense to add some info there
- truncate-command/testrunner still seems to have the wrong comment,
- Untracked files after build git status:
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        Testing/
        core/scripts/bareos-regress.conf
        systemtests/tests/messages/testrunner:
- bareos-check-sources --since-merge:
Plugin 'cmake format' would modify 'core/CTestScript.cmake.in'
Plugin 'clang-format check' would modify 'core/src/dird/ua_status.cc'
Plugin 'trailing newlines' would modify 'systemtests/tests/block-size/etc/bareos/bareos-dir.d/pool/Tape.conf'

**Comment by joergsteffens** [comment]
I applied the requested changes and rebased to master again.

**Review by pstorz — APPROVED** [review]
Good work!

**Pull request merged**